ush/atmos_extractvars.sh

```diff
@@ -0,0 +1,99 @@
+#! /usr/bin/env bash
+################################################################################
+#   Script:    
+#
+source "${USHgfs}/preamble.sh"
+source "${USHgfs}/extractvars_tools.sh"
+fcnt=1
+dcnt=1 
+subdata=${1}
+
+[[ -d "${subdata}" ]] || mkdir -p "${subdata}"
+
+for outtype in "f2d" "f3d"; do
+
+  if [[ "${outtype}" == "f2d" ]];then
+    varlist=${varlist_2d}
+    COMOUT_RFCST_PROD_ATMOS="${COMOUT_RFCST_PROD_ATMOS_F2D}"
+  elif [[ "${outtype}" == "f3d" ]];then 
+    varlist=${varlist_3d}
+    varlist_d=${varlist_3d_d}
+    COMOUT_RFCST_PROD_ATMOS="${COMOUT_RFCST_PROD_ATMOS_F3D}"
+  fi
+
+  outdirpre="${subdata}/${outtype}"
+  [[ -d "${outdirpre}" ]] || mkdir -p "${outdirpre}"
+
+  nh=${FHMIN}
+  while (( nh <= FHMAX_GFS )); do
+    fnh=$(printf "%3.3d" "${nh}")
+
+    if [[ "${outtype}" == "f2d" ]];then
+      if [[ ${nh} -le ${FHMAX_HF_EV} ]];then
+        outres="0p25"
+      else
+        outres="0p50"
+      fi
+    elif [[ "${outtype}" == "f3d" ]];then
+      outres="1p00"
+    fi
+
+    if (( nh < FHMAX_HF_EV )); then
+      outfreq=${FHOUT_HF_GFS}
+    else
+      outfreq=${FHOUT_GFS}
+    fi                                      
+
+    com_var="COMIN_ATMOS_GRIB_${outres}"
+    infile1="${!com_var}/${RUN}.${cycle}.pgrb2.${outres}.f${fnh}"
+    infile2="${!com_var}/${RUN}.${cycle}.pgrb2b.${outres}.f${fnh}"
+    outfile="${outdirpre}/${RUN}.${cycle}.pgrb2.${outres}.f${fnh}"
+    rm -f "${outfile}" #remove outfile if it already exists before extraction
+    requestedvars1="${subdata}/partial_parm1.txt"
+    requestedvars2="${subdata}/partial_parm2.txt"
+    rm -f "${requestedvars1}"
+    rm -f "${requestedvars2}"
+    if [[ ! -f "${requestedvars1}" ]]; then touch "${requestedvars1}"; fi
+    if [[ ! -f "${requestedvars2}" ]]; then touch "${requestedvars2}"; fi
```

Comment:
This should probably be part of the `gen_parmlist()` function (but see comment below). Also, you just removed the existing file, it is probably safe to say it isn't still there unless `rm` failed.